nestkernel/connection_manager.cpp

```diff
@@ -633,13 +638,17 @@ nest::ConnectionManager::connect_( Node& s,
 
   increase_connection_count( tid, syn_id );
 
-  if ( is_primary )
+  if ( not check_primary_connections_[ tid ] and is_primary )
```

Comment:
why do we need `check_primary_connections_` for every thread? wouldn't one for the whole process suffice since threads only read this variable?

Reply:
@jakobj Do you mean checking `has_primary_connections_` instead? I did try without `check_primary_connections_` and did the following
```
if ( not has_primary_connections_ and is_primary )
{
#pragma omp atomic write
    has_primary_connections_ = true;
}
```
but that led to worse performance on Piz Daint. I don't quite understand why.

Reply:
ok, maybe add this as a comment to avoid that someone will "improve" this bit?